File: backend/app/schemas/scheduled_report.py
from datetime import datetime
from typing import List, Literal, Optional

from pydantic import Field

from .common import ContractModel


ScheduledReportType = Literal[
    "pre_market_watchlist",
    "post_market_review",
    "portfolio_daily",
    "news_digest",
]
ScheduledReportRunStatus = Literal["success", "failed", "skipped"]
ScheduledReportTrigger = Literal["manual", "scheduled"]


class ScheduledReportJobUpdate(ContractModel):
    enabled: Optional[bool] = None
    schedule_time: Optional[str] = None
    channel_ids: Optional[List[str]] = None


class ScheduledReportJobRecord(ContractModel):
    report_type: ScheduledReportType
    enabled: bool = False
    schedule_time: str
    channel_ids: List[str] = Field(default_factory=list)
    updated_at: datetime


class ScheduledReportRunRecord(ContractModel):
    id: str
    report_type: ScheduledReportType
    title: str
    summary: str
    body: str
    status: ScheduledReportRunStatus
    trigger: ScheduledReportTrigger
    channel_ids: List[str] = Field(default_factory=list)
    delivered_count: int = 0
    ai_enhanced: bool = False
    ai_provider: Optional[str] = None
    reason: Optional[str] = None
    trading_date: Optional[str] = None
    created_at: datetime

File: frontend/src/types/scheduledReport.ts
export type ScheduledReportType =
  | "pre_market_watchlist"
  | "post_market_review"
  | "portfolio_daily"
  | "news_digest";

export type ScheduledReportRunStatus = "success" | "failed" | "skipped";
export type ScheduledReportTrigger = "manual" | "scheduled";

export interface ScheduledReportJobRecord {
  report_type: ScheduledReportType;
  enabled: boolean;
  schedule_time: string;
  channel_ids: string[];
  updated_at: string;
}

export interface ScheduledReportJobUpdate {
  enabled?: boolean | null;
  schedule_time?: string | null;
  channel_ids?: string[] | null;
}

export interface ScheduledReportRunRecord {
  id: string;
  report_type: ScheduledReportType;
  title: string;
  summary: string;
  body: string;
  status: ScheduledReportRunStatus;
  trigger: ScheduledReportTrigger;
  channel_ids: string[];
  delivered_count: number;
  ai_enhanced: boolean;
  ai_provider?: string | null;
  reason?: string | null;
  trading_date?: string | null;
  created_at: string;
}
